lib/rubocop/cop/lint/empty_in_pattern.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for the presence of `in` pattern branches without a body.
+      #
+      # @example
+      #
+      #   # bad
+      #   case condition
+      #   in [a]
+      #     do_something
+      #   in [a, b]
+      #   end
+      #
+      #   # good
+      #   case condition
+      #   in [a]
+      #     do_something
+      #   in [a, b]
+      #     nil
+      #   end
+      #
+      # @example AllowComments: true (default)
+      #
+      #   # good
+      #   case condition
+      #   in [a]
+      #     do_something
+      #   in [a, b]
+      #     # noop
+      #   end
+      #
+      # @example AllowComments: false
+      #
+      #   # bad
+      #   case condition
+      #   in [a]
+      #     do_something
+      #   in [a, b]
+      #     # noop
+      #   end
+      #
+      class EmptyInPattern < Base
+        extend TargetRubyVersion
+
+        MSG = 'Avoid `in` branches without a body.'
+
+        minimum_target_ruby_version 2.7
+
+        def on_case_match(node)
+          node.in_pattern_branches.each do |branch|
+            next if branch.node_parts[-1] || cop_config['AllowComments'] && comment_lines?(node)
```

Comment:
When https://github.com/rubocop/rubocop-ast/pull/183 can be used, `branch.node_parts[-1]` can be updated to `branch.body`.

Reply:
I updated it.